x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/enrollment/TransportCreateEnrollmentTokenAction.java

```diff
@@ -0,0 +1,141 @@
+            if (caCertificates.size() != 1) {
+                listener.onFailure(new IllegalStateException(
+                    "Unable to create an enrollment token. Elasticsearch node HTTP layer SSL configuration Keystore doesn't contain a " +
+                    "single PrivateKey entry where the associated certificate is a CA certificate"));
+
+            } else {
+                final TimeValue expiration = TimeValue.timeValueSeconds(ENROLL_API_KEY_EXPIRATION_SEC);
+                final List<RoleDescriptor> roleDescriptors = new ArrayList<>(1);
+                final String[] clusterPrivileges = { "enroll" };
+                final RoleDescriptor roleDescriptor = new RoleDescriptor("create_enrollment_token", clusterPrivileges, null, null);
+                roleDescriptors.add(roleDescriptor);
+                CreateApiKeyRequest apiRequest = new CreateApiKeyRequest("enrollment_token_API_key_" + UUIDs.base64UUID(),
+                    roleDescriptors, expiration);
+                final String fingerprint = SslUtil.calculateFingerprint(caCertificates.get(0));
+                createEnrolmentToken(fingerprint, apiRequest, listener);
+            }
+        } catch (Exception e) {
+            listener.onFailure(e);
+        }
+    }
+
+    private void createEnrolmentToken(String fingerprint, CreateApiKeyRequest apiRequest,
+                                      ActionListener<CreateEnrollmentTokenResponse> listener) {
+        try {
+            generator.generateApiKey(securityContext.getAuthentication(), apiRequest,
+                ActionListener.wrap(
+                    CreateApiKeyResponse -> {
+                        final NodeInfo nodeInfo = nodeService.info(false, false, false, false, false, false,
+                            true, false, false, false, false);
+                        final HttpInfo httpInfo = nodeInfo.getInfo(HttpInfo.class);
+                        final String address = httpInfo.getAddress().publishAddress().toString();
+
+                        final XContentBuilder builder = JsonXContent.contentBuilder();
+                        builder.startObject();
+                        builder.field("adr", address);
+                        builder.field("fgr", fingerprint);
+                        builder.field("key", CreateApiKeyResponse.getKey().toString());
+                        builder.endObject();
+                        final String jsonString = Strings.toString(builder);
+                        final String token = Base64.getEncoder().encodeToString(jsonString.getBytes(StandardCharsets.UTF_8));
+
+                        final CreateEnrollmentTokenResponse response = new CreateEnrollmentTokenResponse(token);
+                        listener.onResponse(response);
+                    },
+                    listener::onFailure
+                )
+            );
+        } catch (Exception e) {
+            logger.error(() -> new ParameterizedMessage("Error generating enrollment token"), e);
```

Comment:
using the base class logger is deprecated IIRC, we can get a new Logger instance here.

Reply:
why not simply logger("Error generating enrollment token", e) ?